Objects/StrategyCardAutomator.ttslua

```diff
@@ -1180,6 +1100,14 @@ function _updateColorSelector(layout, element, colorTable)
 	return layout
 end
 
+function onObjectEnterContainer(bag, enter_object)
+	--restore the default UI layout
+	if enter_object.getName() == self.getName() then
+		if _defaultLayoutBackup then
+			UI.setXmlTable(_defaultLayoutBackup)
+		end
+	end
+end
```

Comment:
I'm not sure if this is inefficient, as this function gets called each time an object enters a container.
With the "deleted items" container, this should also cover players trying to delete the object.

Reply:
you can be a little simpler: `if enter_object == self then` without needing getName().

Reply:
moreover, you could instead use onObjectDestroy to also catch the case of someone outright deleting it (it is also called for objects moving into containers, so no need for onObjectEnterContainer).